diff --git a/dist/index.d.mts b/dist/index.d.mts
index afce079c2fbe89883a6ca5c63ae0a9352d1efd5a..3f7afebf982fd23b1b3558c19055607e50d9b035 100644
--- a/dist/index.d.mts
+++ b/dist/index.d.mts
@@ -1683,6 +1683,11 @@ interface CreatePushedAuthorizationRequestOptions {
      * state parameter to use for PAR. If not provided a value will generated automatically
      */
     state?: string;
+    /**
+     * Optional issuer state from the Credential Offer authorization_code grant.
+     * Serialized as issuer_state in the authorization request.
+     */
+    issuerState?: string;
 }
 /**
  * Creates a Pushed Authorization Request (PAR) for OAuth 2.0 authorization flows.
diff --git a/dist/index.d.ts b/dist/index.d.ts
index afce079c2fbe89883a6ca5c63ae0a9352d1efd5a..3f7afebf982fd23b1b3558c19055607e50d9b035 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1683,6 +1683,11 @@ interface CreatePushedAuthorizationRequestOptions {
      * state parameter to use for PAR. If not provided a value will generated automatically
      */
     state?: string;
+    /**
+     * Optional issuer state from the Credential Offer authorization_code grant.
+     * Serialized as issuer_state in the authorization request.
+     */
+    issuerState?: string;
 }
 /**
  * Creates a Pushed Authorization Request (PAR) for OAuth 2.0 authorization flows.
diff --git a/dist/index.js b/dist/index.js
index 999f3238a7f53f42648cac76ff1e38315ca3a01e..a4211e948d119bdbff94784a74813b5e82ae2ee6 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1407,6 +1407,7 @@ async function createPushedAuthorizationRequest(options) {
     client_id: options.clientId,
     code_challenge: pkce.codeChallenge,
     code_challenge_method: pkce.codeChallengeMethod,
+    ...(options.issuerState !== void 0 ? { issuer_state: options.issuerState } : {}),
     jti: options.jti ?? (0, import_io_wallet_utils14.encodeToBase64Url)(
       await options.callbacks.generateRandom(RANDOM_BYTES_SIZE)
     ),
diff --git a/dist/index.mjs b/dist/index.mjs
index 9fde9d49ebd658c643b1985568022c47d6de2b10..8a5bc090e785151a831b757332f2204387dc0eac 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1355,6 +1355,7 @@ async function createPushedAuthorizationRequest(options) {
     client_id: options.clientId,
     code_challenge: pkce.codeChallenge,
     code_challenge_method: pkce.codeChallengeMethod,
+    ...(options.issuerState !== void 0 ? { issuer_state: options.issuerState } : {}),
     jti: options.jti ?? encodeToBase64Url6(
       await options.callbacks.generateRandom(RANDOM_BYTES_SIZE)
     ),
