diff --git a/lib/commonjs/credential/issuance/v1.3.3/01-evaluate-issuer-trust.js b/lib/commonjs/credential/issuance/v1.3.3/01-evaluate-issuer-trust.js
index 4554fd89419d44b5c0478793392f11144d452885..b6e08f15ef830530cd892b372a26a4592e3a1b68 100644
--- a/lib/commonjs/credential/issuance/v1.3.3/01-evaluate-issuer-trust.js
+++ b/lib/commonjs/credential/issuance/v1.3.3/01-evaluate-issuer-trust.js
@@ -13,6 +13,7 @@ const evaluateIssuerTrust = async function (issuerUrl) {
   const issuerMetadata = await (0, _ioWalletOid4vci.fetchMetadata)({
     config: _config.sdkConfigV1_3,
     credentialIssuerUrl: issuerUrl,
+    authorizationServer: context.authorizationServer,
     callbacks: {
       ..._callbacks.partialCallbacks,
       fetch: context.appFetch
diff --git a/lib/module/credential/issuance/v1.3.3/01-evaluate-issuer-trust.js b/lib/module/credential/issuance/v1.3.3/01-evaluate-issuer-trust.js
index cf0c29967a0bbf76e9493ed60bdce2f8ec1399ba..32d67606d4d14879142c60c59e1c9a0f97ade835 100644
--- a/lib/module/credential/issuance/v1.3.3/01-evaluate-issuer-trust.js
+++ b/lib/module/credential/issuance/v1.3.3/01-evaluate-issuer-trust.js
@@ -7,6 +7,7 @@ export const evaluateIssuerTrust = async function (issuerUrl) {
   const issuerMetadata = await fetchMetadata({
     config: sdkConfigV1_3,
     credentialIssuerUrl: issuerUrl,
+    authorizationServer: context.authorizationServer,
     callbacks: {
       ...partialCallbacks,
       fetch: context.appFetch
diff --git a/lib/typescript/credential/issuance/api/01-evaluate-issuer-trust.d.ts b/lib/typescript/credential/issuance/api/01-evaluate-issuer-trust.d.ts
index 05f69dfb38d02808954e65ed9ea04a60c441ba02..8df98e68be7a24655f94b91ab3c99b16ac09748c 100644
--- a/lib/typescript/credential/issuance/api/01-evaluate-issuer-trust.d.ts
+++ b/lib/typescript/credential/issuance/api/01-evaluate-issuer-trust.d.ts
@@ -6,10 +6,12 @@ export interface EvaluateIssuerTrustApi {
      * @since 1.0.0
      *
      * @param issuerUrl The base url of the Issuer
+     * @param context.authorizationServer (optional) Authorization Server URL selected from a credential offer
      * @param context.appFetch (optional) fetch api implementation. Default: built-in fetch
      * @returns The Issuer's configuration
      */
     evaluateIssuerTrust(issuerUrl: string, ctx?: {
+        authorizationServer?: string;
         appFetch?: GlobalFetch["fetch"];
     }): Promise<{
         issuerConf: IssuerConfig;
diff --git a/src/credential/issuance/api/01-evaluate-issuer-trust.ts b/src/credential/issuance/api/01-evaluate-issuer-trust.ts
index 63487e716c8317cc2b941b1ed285452ae89e0f47..36c3a96f34db7fd9afa4d986034275df328221f3 100644
--- a/src/credential/issuance/api/01-evaluate-issuer-trust.ts
+++ b/src/credential/issuance/api/01-evaluate-issuer-trust.ts
@@ -7,11 +7,12 @@ export interface EvaluateIssuerTrustApi {
    * @since 1.0.0
    *
    * @param issuerUrl The base url of the Issuer
+   * @param context.authorizationServer (optional) Authorization Server URL selected from a credential offer
    * @param context.appFetch (optional) fetch api implementation. Default: built-in fetch
    * @returns The Issuer's configuration
    */
   evaluateIssuerTrust(
     issuerUrl: string,
-    ctx?: { appFetch?: GlobalFetch["fetch"] }
+    ctx?: { authorizationServer?: string; appFetch?: GlobalFetch["fetch"] }
   ): Promise<{ issuerConf: IssuerConfig }>;
 }
diff --git a/src/credential/issuance/v1.3.3/01-evaluate-issuer-trust.ts b/src/credential/issuance/v1.3.3/01-evaluate-issuer-trust.ts
index dcaeadcbf5715aa12b7544239d1aaa6871a1cc7d..7977e5d5228514ae5ee4efcb2924d6a693042934 100644
--- a/src/credential/issuance/v1.3.3/01-evaluate-issuer-trust.ts
+++ b/src/credential/issuance/v1.3.3/01-evaluate-issuer-trust.ts
@@ -14,6 +14,7 @@ export const evaluateIssuerTrust: IssuanceApi["evaluateIssuerTrust"] = async (
   const issuerMetadata = (await fetchMetadata({
     config: sdkConfigV1_3,
     credentialIssuerUrl: issuerUrl,
+    authorizationServer: context.authorizationServer,
     callbacks: {
       ...partialCallbacks,
       fetch: context.appFetch,
