diff --git a/src/views/Stack/CardContainer.tsx b/src/views/Stack/CardContainer.tsx
index 1e9ee0e6a5e5b0087cbe2c61146af89f8b050802..36e57bcb131377cf15a3a95d6927958032374cd3 100644
--- a/src/views/Stack/CardContainer.tsx
+++ b/src/views/Stack/CardContainer.tsx
@@ -6,7 +6,7 @@ import {
 } from '@react-navigation/elements';
 import { Route, useTheme } from '@react-navigation/native';
 import * as React from 'react';
-import { Animated, StyleSheet, View } from 'react-native';
+import { AccessibilityInfo, Animated, StyleSheet, View } from 'react-native';
 
 import type { Layout, Scene } from '../../types';
 import ModalPresentationContext from '../../utils/ModalPresentationContext';
@@ -164,6 +164,28 @@ function CardContainer({
   };
 
   const { colors } = useTheme();
+  const [isScreenReaderEnabled, setIsScreenReaderEnabled] =
+    React.useState(false);
+
+  React.useEffect(() => {
+    let isMounted = true;
+
+    AccessibilityInfo.isScreenReaderEnabled().then(enabled => {
+      if (isMounted) {
+        setIsScreenReaderEnabled(enabled);
+      }
+    });
+
+    const subscription = AccessibilityInfo.addEventListener(
+      'screenReaderChanged',
+      setIsScreenReaderEnabled
+    );
+
+    return () => {
+      isMounted = false;
+      subscription.remove();
+    };
+  }, []);
 
   const [pointerEvents, setPointerEvents] = React.useState<'box-none' | 'none'>(
     'box-none'
@@ -197,6 +219,7 @@ function CardContainer({
     gestureVelocityImpact,
     headerMode,
     headerShown,
+    headerTransparent,
     transitionSpec,
   } = scene.descriptor.options;
 
@@ -299,6 +322,14 @@ function CardContainer({
                 getPreviousScene,
                 getFocusedRoute,
                 onContentHeightChange: onHeaderHeightChange,
+                style: [
+                  headerTransparent && isScreenReaderEnabled
+                    ? {
+                        // Specify an explicit min height for Android screen readers
+                        minHeight: headerHeight,
+                      }
+                    : null,
+                ],
               })
             : null}
         </ModalPresentationContext.Provider>
