diff --git a/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java b/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java
index 248ff0825af89ce82e2c7346ef1edc104b06c91c..7a40b165c9cfcbbd21e4bf8c2bd0eaa632ddecf2 100644
--- a/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java
+++ b/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java
@@ -171,6 +171,10 @@ public class RNPushNotificationHelper {
             return;
         }
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU && !getAlarmManager().canScheduleExactAlarms()) {
+            return;
+        }
+
         Log.d(LOG_TAG, String.format("Setting a notification with id %s at time %s",
                 bundle.getString("id"), Long.toString(fireDate)));
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
diff --git a/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java b/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java
index 721ca40cc44a558cbf3cd083849407bcf01ddd79..00b12106723ded1d6e4c0070c82240ebab8f8764 100644
--- a/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java
+++ b/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java
@@ -116,6 +116,17 @@ public class RNReceivedMessageHandler {
                 bundle.putString("largeIconUrl", imageUrl);
             }
         }
+        // GCM legacy handling patch
+        else
+        {
+          Map<String, String> data = message.getData();
+          if(data.containsKey("title") && data.containsKey("message"))
+          {
+            bundle.putString("title", data.get("title"));
+            bundle.putString("message", data.get("message"));
+          }
+        }
+        // End patch
 
         Bundle dataBundle = new Bundle();
         Map<String, String> notificationData = message.getData();
