diff --git a/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm b/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm
index 0d231bc8aa938da296eb3b981e8ac9595a43b87f..042f2a27bf6bb60d98b906e5716697e2ed208aa8 100644
--- a/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm
+++ b/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm
@@ -24,7 +24,6 @@ @interface RCTPullToRefreshViewComponentView () <RCTPullToRefreshViewViewProtoco
 @end
 
 @implementation RCTPullToRefreshViewComponentView {
-  BOOL _isBeforeInitialLayout;
   UIRefreshControl *_refreshControl;
   RCTScrollViewComponentView *__weak _scrollViewComponentView;
 }
@@ -37,7 +36,6 @@ - (instancetype)initWithFrame:(CGRect)frame
     // The pull-to-refresh view is not a subview of this view.
     self.hidden = YES;
 
-    _isBeforeInitialLayout = YES;
     [self _initializeUIRefreshControl];
   }
 
@@ -64,18 +62,11 @@ - (void)prepareForRecycle
   [super prepareForRecycle];
   _scrollViewComponentView = nil;
   _props = nil;
-  _isBeforeInitialLayout = YES;
   [self _initializeUIRefreshControl];
 }
 
 - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &)oldProps
 {
-  // Prop updates are ignored by _refreshControl until after the initial layout, so just store them in _props until then
-  if (_isBeforeInitialLayout) {
-    _props = std::static_pointer_cast<const PullToRefreshViewProps>(props);
-    return;
-  }
-
   const auto &oldConcreteProps = static_cast<const PullToRefreshViewProps &>(*_props);
   const auto &newConcreteProps = static_cast<const PullToRefreshViewProps &>(*props);
 
@@ -111,6 +102,10 @@ - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &
       [_refreshControl endRefreshing];
     }
   }
+
+  if (newConcreteProps.zIndex != oldConcreteProps.zIndex) {
+    _refreshControl.layer.zPosition = newConcreteProps.zIndex.value_or(0);
+  }
 }
 
 #pragma mark -
@@ -155,10 +150,12 @@ - (void)layoutSubviews
 
   // Attempts to begin refreshing before the initial layout are ignored by _refreshControl. So if the control is
   // refreshing when mounted, we need to call beginRefreshing in layoutSubviews or it won't work.
-  if (_isBeforeInitialLayout) {
-    _isBeforeInitialLayout = NO;
+  if (self.window) {
+    const auto &concreteProps = static_cast<const PullToRefreshViewProps &>(*_props);
 
-    [self updateProps:_props oldProps:PullToRefreshViewShadowNode::defaultSharedProps()];
+    if (concreteProps.refreshing) {
+      [self beginRefreshingProgrammatically];
+    }
   }
 }
 
@@ -214,11 +211,12 @@ - (void)beginRefreshingProgrammatically
 
   // When refreshing programmatically (i.e. without pulling down), we must explicitly adjust the ScrollView content
   // offset, or else the _refreshControl won't be visible
-  UIScrollView *scrollView = _scrollViewComponentView.scrollView;
-  CGPoint offset = {scrollView.contentOffset.x, scrollView.contentOffset.y - _refreshControl.frame.size.height};
-  [scrollView setContentOffset:offset];
-
-  [_refreshControl beginRefreshing];
+  if (!_refreshControl.isRefreshing) {
+    UIScrollView *scrollView = _scrollViewComponentView.scrollView;
+    CGPoint offset = {scrollView.contentOffset.x, scrollView.contentOffset.y - _refreshControl.frame.size.height};
+    [scrollView setContentOffset:offset];
+    [_refreshControl beginRefreshing];
+  }
 }
 
 #pragma mark - Native commands
diff --git a/ReactCommon/react/featureflags/ReactNativeFeatureFlagsDefaults.h b/ReactCommon/react/featureflags/ReactNativeFeatureFlagsDefaults.h
index e0742c9c5c97bba076d63d5ab79fcb5356cf0174..a5cd8cfbf10fce0ee43a260134f248401cc0e058 100644
--- a/ReactCommon/react/featureflags/ReactNativeFeatureFlagsDefaults.h
+++ b/ReactCommon/react/featureflags/ReactNativeFeatureFlagsDefaults.h
@@ -216,7 +216,7 @@ class ReactNativeFeatureFlagsDefaults : public ReactNativeFeatureFlagsProvider {
   }
 
   bool preventShadowTreeCommitExhaustion() override {
-    return false;
+    return true;
   }
 
   bool traceTurboModulePromiseRejectionsOnAndroid() override {
