diff --git a/lib/module/createAnimatedComponent/AnimatedComponent.js b/lib/module/createAnimatedComponent/AnimatedComponent.js
index 0245012fbae387969b3c3d6c052db1c76c3fe8e3..b1a88401d6c036a2b2a53def44cb3701b7d70007 100644
--- a/lib/module/createAnimatedComponent/AnimatedComponent.js
+++ b/lib/module/createAnimatedComponent/AnimatedComponent.js
@@ -319,12 +319,16 @@ export default class AnimatedComponent extends ReanimatedAnimatedComponent {
       };
     }
     const skipEntering = this.context?.current;
-    const nativeID = skipEntering ? undefined : `${this.reanimatedID}`;
-    const jestProps = IS_JEST ? {
-      jestInlineStyle: this.props.style && filterOutAnimatedStyles(this.props.style),
-      jestAnimatedStyle: this.jestAnimatedStyle,
-      jestAnimatedProps: this.jestAnimatedProps
-    } : {};
+    let nativeID, jestProps;
+    if (IS_JEST) {
+      jestProps = {
+        jestInlineStyle: this.props.style && filterOutAnimatedStyles(this.props.style),
+        jestAnimatedStyle: this.jestAnimatedStyle,
+        jestAnimatedProps: this.jestAnimatedProps
+      };
+    } else if (!skipEntering) {
+      nativeID = `${this.reanimatedID}`;
+    }
     if (FORCE_REACT_RENDER_FOR_SETTLED_ANIMATIONS) {
       const flatStyles = StyleSheet.flatten(filteredProps.style);
       const mergedStyles = {
