diff --git a/android/build.gradle b/android/build.gradle
index b5d2e83628dd3d67441511b312ff22f2cba43ab5..370d4a61e81231cf81bef717e3f9f4049ffba62d 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -133,7 +133,8 @@ android {
         cppFlags "-O2 -frtti -fexceptions -Wall -Wno-unused-variable -fstack-protector-all"
         arguments "-DANDROID_STL=c++_shared",
                 "-DNODE_MODULES_DIR=${nodeModules}",
-                "-DENABLE_FRAME_PROCESSORS=${enableFrameProcessors ? "ON" : "OFF"}"
+                "-DENABLE_FRAME_PROCESSORS=${enableFrameProcessors ? "ON" : "OFF"}",
+                "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
         abiFilters (*reactNativeArchitectures())
       }
     }
