diff --git a/built/image-zoom/image-zoom.component.d.ts b/built/image-zoom/image-zoom.component.d.ts
index 3f1724bbacd716dd86f693dea6b49d582642e3f7..b033d5e3756e6ea1365cc2030793a98768bfa912 100644
--- a/built/image-zoom/image-zoom.component.d.ts
+++ b/built/image-zoom/image-zoom.component.d.ts
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { LayoutChangeEvent } from 'react-native';
 import { ICenterOn, ImageZoomProps, ImageZoomState } from './image-zoom.type';
-export default class ImageViewer extends React.Component<ImageZoomProps, ImageZoomState> {
+export default class ImageViewer extends React.Component<React.PropsWithChildren<ImageZoomProps>, ImageZoomState> {
     static defaultProps: ImageZoomProps;
     state: ImageZoomState;
     private lastPositionX;
