diff --git a/android/build.gradle b/android/build.gradle
index 6ae1aa71f21669037864495fdc584b839962a46a..23c2f96d3d71ae3a0fb786ea98d8ec57ce51640f 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -49,5 +49,5 @@ dependencies {
     // 1.2.3 is the minimum version compatible with androidx.
     // See https://github.com/uccmawei/FingerprintIdentify/issues/74
     // (translation https://translate.google.com/translate?sl=zh-CN&tl=en&u=https://github.com/uccmawei/FingerprintIdentify/issues/74)
-    implementation "com.github.uccmawei:fingerprintidentify:${safeExtGet("fingerprintidentify", "1.2.6")}"
+    implementation "com.github.uccmawei:FingerprintIdentify:1.2.6"
 }
