diff --git a/node_modules/@gorhom/bottom-sheet/lib/commonjs/components/bottomSheet/BottomSheet.js b/node_modules/@gorhom/bottom-sheet/lib/commonjs/components/bottomSheet/BottomSheet.js
index e5f96e9..857186e 100644
--- a/node_modules/@gorhom/bottom-sheet/lib/commonjs/components/bottomSheet/BottomSheet.js
+++ b/node_modules/@gorhom/bottom-sheet/lib/commonjs/components/bottomSheet/BottomSheet.js
@@ -1255,9 +1255,6 @@ const BottomSheetComponent = /*#__PURE__*/(0, _react.forwardRef)(function Bottom
             }, "BottomSheetBackgroundContainer"), /*#__PURE__*/(0, _jsxRuntime.jsxs)(_reactNativeReanimated.default.View, {
               pointerEvents: "box-none",
               style: contentMaskContainerStyle,
-              accessible: _providedAccessible ?? undefined,
-              accessibilityRole: _providedAccessibilityRole ?? undefined,
-              accessibilityLabel: _providedAccessibilityLabel ?? undefined,
               children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(DraggableView, {
                 style: contentContainerStyle,
                 children: children
diff --git a/node_modules/@gorhom/bottom-sheet/lib/commonjs/components/bottomSheetBackdrop/BottomSheetBackdrop.js b/node_modules/@gorhom/bottom-sheet/lib/commonjs/components/bottomSheetBackdrop/BottomSheetBackdrop.js
index 669fae0..fdf1256 100644
--- a/node_modules/@gorhom/bottom-sheet/lib/commonjs/components/bottomSheetBackdrop/BottomSheetBackdrop.js
+++ b/node_modules/@gorhom/bottom-sheet/lib/commonjs/components/bottomSheetBackdrop/BottomSheetBackdrop.js
@@ -100,10 +100,6 @@ const BottomSheetBackdropComponent = ({
   const AnimatedView = /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNativeReanimated.default.View, {
     style: containerStyle,
     pointerEvents: pointerEvents,
-    accessible: _providedAccessible ?? undefined,
-    accessibilityRole: _providedAccessibilityRole ?? undefined,
-    accessibilityLabel: _providedAccessibilityLabel ?? undefined,
-    accessibilityHint: _providedAccessibilityHint ? _providedAccessibilityHint : `Tap to ${typeof pressBehavior === 'string' ? pressBehavior : 'move'} the Bottom Sheet`,
     children: children
   });
   return pressBehavior !== 'none' ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNativeGestureHandler.GestureDetector, {
diff --git a/node_modules/@gorhom/bottom-sheet/lib/module/components/bottomSheet/BottomSheet.js b/node_modules/@gorhom/bottom-sheet/lib/module/components/bottomSheet/BottomSheet.js
index 5ad2bbd..f9ccc6c 100644
--- a/node_modules/@gorhom/bottom-sheet/lib/module/components/bottomSheet/BottomSheet.js
+++ b/node_modules/@gorhom/bottom-sheet/lib/module/components/bottomSheet/BottomSheet.js
@@ -1247,9 +1247,6 @@ const BottomSheetComponent = /*#__PURE__*/forwardRef(function BottomSheet(props,
             }, "BottomSheetBackgroundContainer"), /*#__PURE__*/_jsxs(Animated.View, {
               pointerEvents: "box-none",
               style: contentMaskContainerStyle,
-              accessible: _providedAccessible ?? undefined,
-              accessibilityRole: _providedAccessibilityRole ?? undefined,
-              accessibilityLabel: _providedAccessibilityLabel ?? undefined,
               children: [/*#__PURE__*/_jsx(DraggableView, {
                 style: contentContainerStyle,
                 children: children
diff --git a/node_modules/@gorhom/bottom-sheet/lib/module/components/bottomSheetBackdrop/BottomSheetBackdrop.js b/node_modules/@gorhom/bottom-sheet/lib/module/components/bottomSheetBackdrop/BottomSheetBackdrop.js
index 395541e..4b00a18 100644
--- a/node_modules/@gorhom/bottom-sheet/lib/module/components/bottomSheetBackdrop/BottomSheetBackdrop.js
+++ b/node_modules/@gorhom/bottom-sheet/lib/module/components/bottomSheetBackdrop/BottomSheetBackdrop.js
@@ -94,10 +94,6 @@ const BottomSheetBackdropComponent = ({
   const AnimatedView = /*#__PURE__*/_jsx(Animated.View, {
     style: containerStyle,
     pointerEvents: pointerEvents,
-    accessible: _providedAccessible ?? undefined,
-    accessibilityRole: _providedAccessibilityRole ?? undefined,
-    accessibilityLabel: _providedAccessibilityLabel ?? undefined,
-    accessibilityHint: _providedAccessibilityHint ? _providedAccessibilityHint : `Tap to ${typeof pressBehavior === 'string' ? pressBehavior : 'move'} the Bottom Sheet`,
     children: children
   });
   return pressBehavior !== 'none' ? /*#__PURE__*/_jsx(GestureDetector, {
diff --git a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx
index a6f048e..d09c8b8 100644
--- a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx
+++ b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheet/BottomSheet.tsx
@@ -1921,9 +1921,6 @@ const BottomSheetComponent = forwardRef<BottomSheet, BottomSheetProps>(
                 <Animated.View
                   pointerEvents="box-none"
                   style={contentMaskContainerStyle}
-                  accessible={_providedAccessible ?? undefined}
-                  accessibilityRole={_providedAccessibilityRole ?? undefined}
-                  accessibilityLabel={_providedAccessibilityLabel ?? undefined}
                 >
                   <DraggableView
                     key="BottomSheetRootDraggableView"
diff --git a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx
index ee10bde..010add4 100644
--- a/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx
+++ b/node_modules/@gorhom/bottom-sheet/src/components/bottomSheetBackdrop/BottomSheetBackdrop.tsx
@@ -139,16 +139,6 @@ const BottomSheetBackdropComponent = ({
     <Animated.View
       style={containerStyle}
       pointerEvents={pointerEvents}
-      accessible={_providedAccessible ?? undefined}
-      accessibilityRole={_providedAccessibilityRole ?? undefined}
-      accessibilityLabel={_providedAccessibilityLabel ?? undefined}
-      accessibilityHint={
-        _providedAccessibilityHint
-          ? _providedAccessibilityHint
-          : `Tap to ${
-              typeof pressBehavior === 'string' ? pressBehavior : 'move'
-            } the Bottom Sheet`
-      }
     >
       {children}
     </Animated.View>
