diff --git a/node_modules/@react-native-community/cli/build/commands/.DS_Store b/node_modules/@react-native-community/cli/build/commands/.DS_Store
new file mode 100644
index 0000000..e097ea7
Binary files /dev/null and b/node_modules/@react-native-community/cli/build/commands/.DS_Store differ
diff --git a/node_modules/@react-native-community/cli/build/commands/runIOS/findMatchingSimulator.js b/node_modules/@react-native-community/cli/build/commands/runIOS/findMatchingSimulator.js
index 4f75b7a..469c112 100644
--- a/node_modules/@react-native-community/cli/build/commands/runIOS/findMatchingSimulator.js
+++ b/node_modules/@react-native-community/cli/build/commands/runIOS/findMatchingSimulator.js
@@ -65,7 +65,7 @@ function findMatchingSimulator(simulators, simulatorString) {
     for (const i in device) {
       const simulator = device[i]; // Skipping non-available simulator
 
-      if (simulator.availability !== '(available)' && simulator.isAvailable !== 'YES') {
+      if (simulator.availability !== '(available)' && simulator.isAvailable !== 'YES' && simulator.isAvailable !== true) {
         continue;
       }
 
diff --git a/node_modules/@react-native-community/cli/build/commands/server/external/xsel b/node_modules/@react-native-community/cli/build/commands/server/external/xsel
old mode 100644
new mode 100755
