diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index 5a65d89..b9bf0bf 100644
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -8284,6 +8284,10 @@ export interface ToastAndroidStatic {
 }
 
 export interface UIManagerStatic {
+    //Added missing accesibility definition
+    sendAccessibilityEvent(reactTag?: number, eventType: number): void;
+    AccessibilityEventTypes: { typeViewFocused : number; }
+    
     /**
      * Capture an image of the screen, window or an individual view. The image
      * will be stored in a temporary file that will only exist for as long as the
