diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index bf4b3d6..a3471e9 100755
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -8370,6 +8370,10 @@ export interface UIManagerStatic {
      * commandArgs - Args of the native method that we can pass from JS to native.
      */
     dispatchViewManagerCommand: (reactTag: number | null, commandID: number | string, commandArgs?: Array<any>) => void;
+
+    //Added missing accesibility definition
+    sendAccessibilityEvent(reactTag?: number, eventType: number): void;
+    AccessibilityEventTypes: { typeViewFocused : number; }
 }
 
 export interface SwitchPropsIOS extends ViewProps {
