diff --git a/node_modules/@types/react-native-text-input-mask/index.d.ts b/node_modules/@types/react-native-text-input-mask/index.d.ts
index 4cf4b39..8eacdbc 100644
--- a/node_modules/@types/react-native-text-input-mask/index.d.ts
+++ b/node_modules/@types/react-native-text-input-mask/index.d.ts
@@ -4,6 +4,7 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
+import {MutableRefObject} from "react";
 import * as React from "react";
 import * as ReactNative from "react-native";
 
@@ -221,6 +222,10 @@ export interface TextInputMaskProps extends ReactNative.ViewProps, ReactNative.T
      * or set/update maxLength to prevent unwanted edits without flicker.
      */
     value?: string;
+
+    // Patch: Adding missing types ref and isNumericSecureKeyboard
+    ref?: ((instance: TextInputMask | null) => void) | MutableRefObject<TextInputMask | null> | null;
+    isNumericSecureKeyboard?: boolean,
 }
 
 export default class TextInputMask extends React.Component<TextInputMaskProps> { }
