diff --git a/node_modules/@types/react-native-text-input-mask/index.d.ts b/node_modules/@types/react-native-text-input-mask/index.d.ts
index 4cf4b39..cdb9713 100644
--- a/node_modules/@types/react-native-text-input-mask/index.d.ts
+++ b/node_modules/@types/react-native-text-input-mask/index.d.ts
@@ -180,6 +180,11 @@ export interface TextInputMaskProps extends ReactNative.ViewProps, ReactNative.T
      */
     secureTextEntry?: boolean;
 
+    /**
+     * If true, a numeric keypad will be shown and the numbers entered will be obscures
+     */
+    isNumericSecureKeyboard?: boolean;
+
     /**
      * If true, all text will automatically be selected on focus
      */
