diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index d945df7..6eb1dbd 100644
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -6928,6 +6928,7 @@ type AccessibilityEvent = AccessibilityChangeEvent | AccessibilityAnnoucementFin
  * @see https://facebook.github.io/react-native/docs/accessibilityinfo.html
  */
 export interface AccessibilityInfoStatic {
+	isScreenReaderEnabled: () => Promise<boolean>;
     /**
      * Query whether a screen reader is currently enabled.
      * Returns a promise which resolves to a boolean. The result is true when a screen reader is enabled and false otherwise.
@@ -8329,6 +8330,10 @@ export interface ToastAndroidStatic {
 }
 
 export interface UIManagerStatic {
+	sendAccessibilityEvent(nodeRef: number, type: any);
+  	AccessibilityEventTypes: {
+    	typeViewFocused : any;
+  	}
     /**
      * Capture an image of the screen, window or an individual view. The image
      * will be stored in a temporary file that will only exist for as long as the
