diff --git a/node_modules/@types/react-native/index.d.ts b/node_modules/@types/react-native/index.d.ts
index a8601d0..45719a2 100644
--- a/node_modules/@types/react-native/index.d.ts
+++ b/node_modules/@types/react-native/index.d.ts
@@ -7789,6 +7789,8 @@ export interface ToastAndroidStatic {
 }
 
 export interface UIManagerStatic {
+    sendAccessibilityEvent(nodeRef: number, type: any);
+    AccessibilityEventTypes: { typeViewFocused : any; }
     /**
      * Capture an image of the screen, window or an individual view. The image
      * will be stored in a temporary file that will only exist for as long as the
