diff --git a/node_modules/danger-plugin-digitalcitizenship/dist/utils.js b/node_modules/danger-plugin-digitalcitizenship/dist/utils.js
index 82e1d17..3d9a783 100644
--- a/node_modules/danger-plugin-digitalcitizenship/dist/utils.js
+++ b/node_modules/danger-plugin-digitalcitizenship/dist/utils.js
@@ -6,7 +6,7 @@ const Pivotal = require("pivotaljs");
  * see https://www.pivotaltracker.com/help/articles/githubs_service_hook_for_tracker/
  */
 function getPivotalStoryIDs(message) {
-    const matches = message.match(/^\[(#\d+(,#\d+)*)\]\s.+/);
+    const matches = message.match(/\[(#\d+(,#\d+)*)\]\s.+/);
     if (matches) {
         return matches[1]
             .split(",")
diff --git a/node_modules/danger-plugin-digitalcitizenship/src/utils.ts b/node_modules/danger-plugin-digitalcitizenship/src/utils.ts
index 5037cb8..7e815d9 100644
--- a/node_modules/danger-plugin-digitalcitizenship/src/utils.ts
+++ b/node_modules/danger-plugin-digitalcitizenship/src/utils.ts
@@ -5,7 +5,7 @@ import Pivotal = require("pivotaljs");
  * see https://www.pivotaltracker.com/help/articles/githubs_service_hook_for_tracker/
  */
 export function getPivotalStoryIDs(message: string): ReadonlyArray<string> {
-  const matches = message.match(/^\[(#\d+(,#\d+)*)\]\s.+/);
+  const matches = message.match(/\[(#\d+(,#\d+)*)\]\s.+/);
   if(matches) {
     return matches[1]
       .split(",")
