diff --git a/node_modules/instabug-reactnative/index.d.ts b/node_modules/instabug-reactnative/index.d.ts
index 56954a8..37545ed 100644
--- a/node_modules/instabug-reactnative/index.d.ts
+++ b/node_modules/instabug-reactnative/index.d.ts
@@ -9,7 +9,7 @@ export namespace BugReporting {
   function setInvocationOptions(invocationOptions: invocationOptions[]): void;
   function onInvokeHandler(handler: () => void): void;
   function onReportSubmitHandler(preSendingHandler: () => void): void;
-  function onSDKDismissedHandler(handler: (dismiss: dismissType, report: reportType) => void): void;
+  function onSDKDismissedHandler(handler: (dismiss: string, report: string) => void): void;
   function setDidSelectPromptOptionHandler(didSelectPromptOptionHandler: () => void): void;
   function setShakingThresholdForiPhone(iPhoneShakingThreshold: number): void;
   function setShakingThresholdForiPad(iPadShakingThreshold: number): void;
@@ -80,7 +80,7 @@ export namespace Replies {
   function show(): void;
   function setOnNewReplyReceivedCallback(onNewReplyReceivedCallback: () => void): void;
   function setOnNewReplyReceivedHandler(onNewReplyReceivedHandler: () => void): void;
-  function getUnreadRepliesCount(messageCountCallback: (count: number) => void): void;
+  function getUnreadRepliesCount(unreadRepliesCountCallback: (count: number) => void): void;
   function setInAppNotificationsEnabled(inAppNotificationsEnabled: boolean): void;
   function setInAppNotificationSound(shouldPlaySound: boolean): void;
 }
@@ -126,7 +126,7 @@ export function setIBGLogPrintsToConsole(printsToConsole: boolean): void;
 export function setCrashReportingEnabled(enableCrashReporter: boolean): void;
 export function setDidSelectPromptOptionHandler(didSelectPromptOptionHandler: () => void): void;
 export function setSessionProfilerEnabled(sessionProfilerEnabled: boolean): void;
-export function getUnreadMessagesCount(messageCountCallback: (count: number) => void): void;
+export function getUnreadMessagesCount(unreadMessagesCountCallback: (count: number) => void): void;
 export function setPushNotificationsEnabled(isPushNotificationEnabled: boolean): void;
 export function setEmailFieldRequiredForActions(
   isEmailFieldRequired: boolean,
