diff --git a/node_modules/instabug-reactnative/index.d.ts b/node_modules/instabug-reactnative/index.d.ts
index 09ec669..04e9e94 100644
--- a/node_modules/instabug-reactnative/index.d.ts
+++ b/node_modules/instabug-reactnative/index.d.ts
@@ -14,7 +14,7 @@ export namespace BugReporting {
     ): void;
   function onInvokeHandler(preInvocationHandler: () => void): void;
   function onReportSubmitHandler(preSendingHandler: () => void): void;
-  function onSDKDismissedHandler(postInvocationHandler: (dismiss: dismissType, report: reportType) => void): void;
+  function onSDKDismissedHandler(postInvocationHandler: (dismiss: string, report: string) => void): void;
   function setPromptOptionsEnabled(
     chat: boolean,
     bug: boolean,
@@ -152,7 +152,7 @@ export function setFloatingButtonEdge(
   ): void;
 export function setLocale(locale: locale): void;
 export function setColorTheme(colorTheme: colorTheme): void;
-export function setPrimaryColor(setPrimaryColor: number): void;
+export function setPrimaryColor(setPrimaryColor: string): void;
 export function appendTags(tags: string[]): void;
 export function resetTags(): void;
 export function getTags(tagsCallback: () => void): void;
