diff --git a/node_modules/native-base/index.d.ts b/node_modules/native-base/index.d.ts
index 7cf347d..58cf81d 100644
--- a/node_modules/native-base/index.d.ts
+++ b/node_modules/native-base/index.d.ts
@@ -366,6 +366,7 @@ declare module "native-base" {
              */
 			androidRippleColor?: string;
 			touchableHighlightStyle?: RnViewStyleProp;
+			underlayColor?: string;
 		}

 		interface Separator extends Testable {
