diff --git a/node_modules/react-native-pdf/ios/RCTPdf/RCTPdfView.m b/node_modules/react-native-pdf/ios/RCTPdf/RCTPdfView.m
index 52aafd8..5efda0b 100644
--- a/node_modules/react-native-pdf/ios/RCTPdf/RCTPdfView.m
+++ b/node_modules/react-native-pdf/ios/RCTPdf/RCTPdfView.m
@@ -188,12 +188,16 @@ const float MIN_SCALE = 1.0f;
         }
 
         if (_pdfDocument && ([changedProps containsObject:@"path"] || [changedProps containsObject:@"enableAnnotationRendering"])) {
-            if (!_enableAnnotationRendering) {
-                for (unsigned long i=0; i<_pdfView.document.pageCount; i++) {
-                    PDFPage *pdfPage = [_pdfView.document pageAtIndex:i];
-                    for (unsigned long j=0; j<pdfPage.annotations.count; j++) {
-                        pdfPage.annotations[j].shouldDisplay = _enableAnnotationRendering;
-                    }
+            for (unsigned long i=0; i<_pdfView.document.pageCount; i++) {
+                PDFPage *pdfPage = [_pdfView.document pageAtIndex:i];
+                for (unsigned long j=0; j<pdfPage.annotations.count; j++) {
+                  if (_enableAnnotationRendering) {
+                    // Annotations are read-only by default on Android so we have to disable them here on iOS
+                    pdfPage.annotations[j].readOnly = true;
+                  } else {
+                    // If annotations are hidden, there is no need to flag them as read-only, just hide them
+                    pdfPage.annotations[j].shouldDisplay = false;
+                  }
                 }
             }
         }
