diff --git a/node_modules/react-native-popup-menu/src/helpers.js b/node_modules/react-native-popup-menu/src/helpers.js
index 880c239..2115168 100644
--- a/node_modules/react-native-popup-menu/src/helpers.js
+++ b/node_modules/react-native-popup-menu/src/helpers.js
@@ -27,7 +27,7 @@ export const makeName = (function() {
  */
 export function makeTouchable(TouchableComponent) {
   const Touchable = TouchableComponent || Platform.select({
-    android: TouchableNativeFeedback,
+    android: Platform.version >= 21 ? TouchableNativeFeedback : TouchableHighlight,
     ios: TouchableHighlight,
     default: TouchableHighlight,
   });
