diff --git a/node_modules/react-native/React/AccessibilityResources/it.lproj/Localizable.strings b/node_modules/react-native/React/AccessibilityResources/it.lproj/Localizable.strings
new file mode 100644
index 0000000..f09950f
--- /dev/null
+++ b/node_modules/react-native/React/AccessibilityResources/it.lproj/Localizable.strings
@@ -0,0 +1,26 @@
+/*
+  Localizable.strings
+  React
+*/
+"alert"="alert";
+"checkbox"="checkbox";
+"combobox"="combo box";
+"menu"="menu";
+"menubar"="menu bar";
+"menuitem"="menu item";
+"progressbar"="progress bar";
+"radio"="radio button";
+"radiogroup"="radio group";
+"scrollbar"="scroll bar";
+"spinbutton"="spin button";
+"switch"="switch";
+"tab"="tab description";
+"tablist"="tab list";
+"timer"="timer";
+"toolbar"="barra degli strumenti";
+"checked"="selezionato";
+"unchecked"="non selezionato";
+"busy"="occupato";
+"expanded"="expanded";
+"collapsed"="collapsed";
+"mixed"="mixed";
diff --git a/node_modules/react-native/scripts/react_native_pods.rb b/node_modules/react-native/scripts/react_native_pods.rb
index db9a34a..e95b882 100644
--- a/node_modules/react-native/scripts/react_native_pods.rb
+++ b/node_modules/react-native/scripts/react_native_pods.rb
@@ -165,7 +165,7 @@ def use_react_native_codegen!(spec, options={})
   return if ENV['DISABLE_CODEGEN'] == '1'
 
   # The path to react-native (e.g. react_native_path)
-  prefix = options[:path] ||= File.join(__dir__, "..")
+  prefix = options[:path] ||= "../../node_modules/react-native"
 
   # The path to JavaScript files
   srcs_dir = options[:srcs_dir] ||= File.join(prefix, "Libraries")
@@ -208,12 +208,12 @@ def use_react_native_codegen!(spec, options={})
     :name => 'Generate Specs',
     :input_files => [srcs_dir],
     :output_files => ["$(DERIVED_FILE_DIR)/codegen-#{codegen_modules_library_name}.log"].concat(generated_files),
-    :script => "set -o pipefail\n\nbash -l -c '#{env_vars} CODEGEN_MODULES_LIBRARY_NAME=#{codegen_modules_library_name} #{File.join(__dir__, "generate-specs.sh")}' 2>&1 | tee \"${SCRIPT_OUTPUT_FILE_0}\"",
+    :script => "set -o pipefail\n\nbash -l -c '#{env_vars} CODEGEN_MODULES_LIBRARY_NAME=#{codegen_modules_library_name} ../../node_modules/react-native/scripts/generate-specs.sh' 2>&1 | tee \"${SCRIPT_OUTPUT_FILE_0}\"",
     :execution_position => :before_compile,
     :show_env_vars_in_log => true
   }
 
-  spec.prepare_command = "#{mkdir_command} && touch #{generated_files.reduce() { |str, file| str + " " + file }}"
+    spec.prepare_command = "cd ../.. && #{mkdir_command} && touch #{generated_files.reduce() { |str, file| str + " " + file }}"
 end
 
 # Local method for the Xcode 12.5 fix
