diff --git a/node_modules/react-native-image-pan-zoom/built/image-zoom/image-zoom.component.d.ts b/node_modules/react-native-image-pan-zoom/built/image-zoom/image-zoom.component.d.ts
index 3f1724b..b033d5e 100644
--- a/node_modules/react-native-image-pan-zoom/built/image-zoom/image-zoom.component.d.ts
+++ b/node_modules/react-native-image-pan-zoom/built/image-zoom/image-zoom.component.d.ts
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { LayoutChangeEvent } from 'react-native';
 import { ICenterOn, ImageZoomProps, ImageZoomState } from './image-zoom.type';
-export default class ImageViewer extends React.Component<ImageZoomProps, ImageZoomState> {
+export default class ImageViewer extends React.Component<React.PropsWithChildren<ImageZoomProps>, ImageZoomState> {
     static defaultProps: ImageZoomProps;
     state: ImageZoomState;
     private lastPositionX;
