diff --git a/node_modules/react-native-device-info/lib/typescript/internal/privateTypes.d.ts b/node_modules/react-native-device-info/lib/typescript/internal/privateTypes.d.ts
index 7ca9a81..9c274ab 100644
--- a/node_modules/react-native-device-info/lib/typescript/internal/privateTypes.d.ts
+++ b/node_modules/react-native-device-info/lib/typescript/internal/privateTypes.d.ts
@@ -79,12 +79,8 @@ interface ExposedNativeMethods {
     getInstallReferrerSync: () => string;
     getInstanceId: () => Promise<string>;
     getInstanceIdSync: () => string;
-    getIpAddress: () => Promise<string>;
-    getIpAddressSync: () => string;
     getLastUpdateTime: () => Promise<number>;
     getLastUpdateTimeSync: () => number;
-    getMacAddress: () => Promise<string>;
-    getMacAddressSync: () => string;
     getMaxMemory: () => Promise<number>;
     getMaxMemorySync: () => number;
     getPhoneNumber: () => Promise<string>;
diff --git a/node_modules/react-native-device-info/src/index.ts b/node_modules/react-native-device-info/src/index.ts
index b298fd9..c611f04 100644
--- a/node_modules/react-native-device-info/src/index.ts
+++ b/node_modules/react-native-device-info/src/index.ts
@@ -58,13 +58,6 @@ export const [getAndroidId, getAndroidIdSync] = getSupportedPlatformInfoFunction
   defaultValue: 'unknown',
 });
 
-export const [getIpAddress, getIpAddressSync] = getSupportedPlatformInfoFunctions({
-  supportedPlatforms: ['android', 'ios', 'windows'],
-  getter: () => RNDeviceInfo.getIpAddress(),
-  syncGetter: () => RNDeviceInfo.getIpAddressSync(),
-  defaultValue: 'unknown',
-});
-
 export const [isCameraPresent, isCameraPresentSync] = getSupportedPlatformInfoFunctions({
   supportedPlatforms: ['android', 'windows', 'web'],
   getter: () => RNDeviceInfo.isCameraPresent(),
@@ -72,23 +65,6 @@ export const [isCameraPresent, isCameraPresentSync] = getSupportedPlatformInfoFu
   defaultValue: false,
 });
 
-export async function getMacAddress() {
-  if (Platform.OS === 'android') {
-    return RNDeviceInfo.getMacAddress();
-  } else if (Platform.OS === 'ios') {
-    return '02:00:00:00:00:00';
-  }
-  return 'unknown';
-}
-
-export function getMacAddressSync() {
-  if (Platform.OS === 'android') {
-    return RNDeviceInfo.getMacAddressSync();
-  } else if (Platform.OS === 'ios') {
-    return '02:00:00:00:00:00';
-  }
-  return 'unknown';
-}
 
 export const getDeviceId = () =>
   getSupportedPlatformInfoSync({
@@ -823,12 +799,8 @@ const deviceInfoModule: DeviceInfoModule = {
   getInstallReferrerSync,
   getInstanceId,
   getInstanceIdSync,
-  getIpAddress,
-  getIpAddressSync,
   getLastUpdateTime,
   getLastUpdateTimeSync,
-  getMacAddress,
-  getMacAddressSync,
   getManufacturer,
   getManufacturerSync,
   getMaxMemory,
diff --git a/node_modules/react-native-device-info/src/internal/privateTypes.ts b/node_modules/react-native-device-info/src/internal/privateTypes.ts
index c345201..3032eb8 100644
--- a/node_modules/react-native-device-info/src/internal/privateTypes.ts
+++ b/node_modules/react-native-device-info/src/internal/privateTypes.ts
@@ -84,12 +84,8 @@ interface ExposedNativeMethods {
   getInstallReferrerSync: () => string;
   getInstanceId: () => Promise<string>;
   getInstanceIdSync: () => string;
-  getIpAddress: () => Promise<string>;
-  getIpAddressSync: () => string;
   getLastUpdateTime: () => Promise<number>;
   getLastUpdateTimeSync: () => number;
-  getMacAddress: () => Promise<string>;
-  getMacAddressSync: () => string;
   getMaxMemory: () => Promise<number>;
   getMaxMemorySync: () => number;
   getPhoneNumber: () => Promise<string>;
