diff --git a/node_modules/react-native-flag-secure-android/.DS_Store b/node_modules/react-native-flag-secure-android/.DS_Store
new file mode 100644
index 0000000..8186bc2
Binary files /dev/null and b/node_modules/react-native-flag-secure-android/.DS_Store differ
diff --git a/node_modules/react-native-flag-secure-android/android/build.gradle b/node_modules/react-native-flag-secure-android/android/build.gradle
index 88fa0af..242e219 100644
--- a/node_modules/react-native-flag-secure-android/android/build.gradle
+++ b/node_modules/react-native-flag-secure-android/android/build.gradle
@@ -36,5 +36,5 @@ android {
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+'
+    implementation 'com.facebook.react:react-native:+'
 }
