diff --git a/node_modules/react-native-screen-brightness/android/build.gradle b/node_modules/react-native-screen-brightness/android/build.gradle
index 97e0d44..3dd14ce 100644
--- a/node_modules/react-native-screen-brightness/android/build.gradle
+++ b/node_modules/react-native-screen-brightness/android/build.gradle
@@ -35,5 +35,5 @@ repositories {
 
 dependencies {
     implementation 'com.facebook.react:react-native:+'
-    implementation 'androidx.core:core:1.+'
+    implementation 'androidx.core:core:1.5.0'
 }
