diff --git a/node_modules/react-native-mixpanel/.DS_Store b/node_modules/react-native-mixpanel/.DS_Store
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/react-native-mixpanel/RNMixpanel/RNMixpanel.m b/node_modules/react-native-mixpanel/RNMixpanel/RNMixpanel.m
index 8564911..8a158db 100644
--- a/node_modules/react-native-mixpanel/RNMixpanel/RNMixpanel.m
+++ b/node_modules/react-native-mixpanel/RNMixpanel/RNMixpanel.m
@@ -46,7 +46,12 @@ -(Mixpanel*) getInstance: (NSString *)name {
                                                   trackCrashes:trackCrashes
                                          automaticPushTracking:automaticPushTracking
                                        optOutTrackingByDefault:optOutTrackingByDefault];
-
+        // disable A/B testing see https://pagopa.atlassian.net/browse/IA-42
+        [instance setEnableVisualABTestAndCodeless: NO];
+        instance.enableVisualABTestAndCodeless=NO;
+        // need to change mixpanel endpoint due to saving data in EU zone
+        // see https://www.pivotaltracker.com/story/show/171600487
+        instance.serverURL = @"https://api-eu.mixpanel.com";
         // copy instances and add the new instance.  then reassign instances
         NSMutableDictionary *newInstances = [NSMutableDictionary dictionaryWithDictionary:instances];
         [newInstances setObject:instance forKey:apiToken];
diff --git a/node_modules/react-native-mixpanel/android/.project b/node_modules/react-native-mixpanel/android/.project
new file mode 100644
index 0000000..95512e0
--- /dev/null
+++ b/node_modules/react-native-mixpanel/android/.project
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-mixpanel</name>
+	<comment>Project react-native-mixpanel created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+</projectDescription>
diff --git a/node_modules/react-native-mixpanel/android/src/main/java/com/kevinejohn/RNMixpanel/RNMixpanelModule.java b/node_modules/react-native-mixpanel/android/src/main/java/com/kevinejohn/RNMixpanel/RNMixpanelModule.java
index 6239e11..a7352d4 100644
--- a/node_modules/react-native-mixpanel/android/src/main/java/com/kevinejohn/RNMixpanel/RNMixpanelModule.java
+++ b/node_modules/react-native-mixpanel/android/src/main/java/com/kevinejohn/RNMixpanel/RNMixpanelModule.java
@@ -18,7 +18,7 @@ import org.json.JSONObject;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
-
+import com.mixpanel.android.mpmetrics.MPConfig;
 /**
  * Mixpanel React Native module.
  * Note that synchronized(instance) is used in methods because that's what MixpanelAPI.java recommends you do if you are keeping instances.
@@ -30,6 +30,13 @@ public class RNMixpanelModule extends ReactContextBaseJavaModule implements Life
     public RNMixpanelModule(ReactApplicationContext reactContext) {
         super(reactContext);
 
+        
+
+        // need to change mixpanel endpoint due to saving data in EU zone
+        // see https://www.pivotaltracker.com/story/show/171600487
+        MPConfig.getInstance(reactContext).setEventsEndpoint("https://api-eu.mixpanel.com/track");
+        MPConfig.getInstance(reactContext).setPeopleEndpoint("https://api-eu.mixpanel.com/engage");
+        MPConfig.getInstance(reactContext).setGroupsEndpoint("https://api-eu.mixpanel.com/groups");
         // Get lifecycle notifications to flush mixpanel on pause or destroy
         reactContext.addLifecycleEventListener(this);
     }
diff --git a/node_modules/react-native-mixpanel/index.d.ts b/node_modules/react-native-mixpanel/index.d.ts
index c00ae8c..df23b29 100644
--- a/node_modules/react-native-mixpanel/index.d.ts
+++ b/node_modules/react-native-mixpanel/index.d.ts
@@ -5,7 +5,7 @@ declare module 'react-native-mixpanel' {
     initialize(): Promise<void>
     getDistinctId(): Promise<string>
     getSuperProperty(propertyName: string): Promise<any>
-    track(event: string, properties?: Object): Promise<void>
+    track(event: string, properties?: {[key: string]: unknown}): Promise<void>
     flush(): Promise<void>
     disableIpAddressGeolocalization(): Promise<void>
     alias(alias: string, oldDistinctID?: string): Promise<void>
