diff --git a/node_modules/react-native-push-notification/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java b/node_modules/react-native-push-notification/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java
index b255f89..eb39e01 100644
--- a/node_modules/react-native-push-notification/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java
+++ b/node_modules/react-native-push-notification/android/src/main/java/com/dieam/reactnativepushnotification/modules/RNReceivedMessageHandler.java
@@ -110,6 +110,17 @@ public class RNReceivedMessageHandler {
                 bundle.putString("largeIconUrl", imageUrl);
             }
         }
+        // GCM legacy handling patch
+        else
+        {
+          Map<String, String> data = message.getData();
+          if(data.containsKey("title") && data.containsKey("message"))
+          {
+            bundle.putString("title", data.get("title"));
+            bundle.putString("message", data.get("message"));
+          }
+        }
+        // End patch
 
         Bundle dataBundle = new Bundle();
         Map<String, String> notificationData = message.getData();
