diff --git a/node_modules/react-native-maps/index.d.ts b/node_modules/react-native-maps/index.d.ts
index d8f55d5..17ed942 100644
--- a/node_modules/react-native-maps/index.d.ts
+++ b/node_modules/react-native-maps/index.d.ts
@@ -33,12 +33,24 @@ declare module "react-native-maps" {
     y: number;
   }
 
+  export type EventActionType =
+    | 'marker-press'
+    | 'polygon-press'
+    | 'polyline-press'
+    | 'callout-press'
+    | 'press'
+    | 'long-press'
+    | 'overlay-press'
+    | undefined;
+
   // helper interface
   export interface MapEvent<T = {}>
     extends NativeSyntheticEvent<
       T & {
         coordinate: LatLng;
         position: Point;
+        action: EventActionType;
+        id?: string;
       }
     > {}
 
