diff --git a/node_modules/react-native/React/AccessibilityResources/it.lproj/Localizable.strings b/node_modules/react-native/React/AccessibilityResources/it.lproj/Localizable.strings
new file mode 100644
index 0000000..f09950f
--- /dev/null
+++ b/node_modules/react-native/React/AccessibilityResources/it.lproj/Localizable.strings
@@ -0,0 +1,26 @@
+/*
+  Localizable.strings
+  React
+*/
+"alert"="alert";
+"checkbox"="casella di controllo";
+"combobox"="elenco a discesa";
+"menu"="menu";
+"menubar"="barra del menu";
+"menuitem"="voce del menu";
+"progressbar"="barra di avanzamento";
+"radio"="pulsante di scelta";
+"radiogroup"="scelta multipla";
+"scrollbar"="barra di scorrimento";
+"spinbutton"="spin button";
+"switch"="switch";
+"tab"="tab description";
+"tablist"="tab list";
+"timer"="timer";
+"toolbar"="barra degli strumenti";
+"checked"="selezionato";
+"unchecked"="non selezionato";
+"busy"="occupato";
+"expanded"="espanso";
+"collapsed"="compresso";
+"mixed"="misto";
