patch-package
--- a/node_modules/react-native-shortcutbadger/android/src/main/java/com/yangqiong/shortcutbadger/ShortcutBadgerPackage.java
+++ b/node_modules/react-native-shortcutbadger/android/src/main/java/com/yangqiong/shortcutbadger/ShortcutBadgerPackage.java
@@ -14,7 +14,6 @@ import java.util.List;
  * Created by yang on 16/8/17.
  */
 public class ShortcutBadgerPackage implements ReactPackage {
-    @Override
     public List<Class<? extends JavaScriptModule>> createJSModules() {
         return Collections.emptyList();
     }
