diff --git a/node_modules/react-native-safe-area-view/index.js b/node_modules/react-native-safe-area-view/index.js
index 0f93e5a..8ecce5c 100644
--- a/node_modules/react-native-safe-area-view/index.js
+++ b/node_modules/react-native-safe-area-view/index.js
@@ -174,6 +174,12 @@ class SafeView extends Component {
     const WIDTH = isLandscape ? X_HEIGHT : X_WIDTH;
     const HEIGHT = isLandscape ? X_WIDTH : X_HEIGHT;
 
+    // temporary fix, in order to execute the application after the porting to 0.63.x, waiting to replace react-navigation
+    if(this.view._component === undefined)
+    {
+      return;
+    }
+
     this.view._component.measureInWindow((winX, winY, winWidth, winHeight) => {
       if (!this.view) {
         return;
